examples/sac/sac.py

```diff
@@ -2,193 +2,173 @@
+    rewards = []
+    rewards_eval = []
+
+    # Main loop
+    collected_frames = 0
+    pbar = tqdm.tqdm(total=cfg.total_frames)
+    r0 = None
+    q_loss = None
+
+    for i, tensordict in enumerate(collector):
+        # update weights of the inference policy
+        collector.update_policy_weights_()
+
+        if r0 is None:
+            r0 = tensordict["next", "reward"].sum(-1).mean().item()
+        pbar.update(tensordict.numel())
+
+        tensordict = tensordict.view(-1)
+        current_frames = tensordict.numel()
+        replay_buffer.extend(tensordict.cpu())
+        collected_frames += current_frames
+
+        # optimization steps
+        if collected_frames >= cfg.init_random_frames:
+            (actor_losses, q_losses, alpha_losses) = ([], [], [])
+            for _ in range(
+                int(cfg.env_per_collector * cfg.frames_per_batch * cfg.utd_ratio)
+            ):
+                # sample from replay buffer
+                sampled_tensordict = replay_buffer.sample(cfg.batch_size).clone()
+
+                loss_td = loss_module(sampled_tensordict)
+
+                actor_loss = loss_td["loss_actor"]
+                q_loss = loss_td["loss_qvalue"]
+                alpha_loss = loss_td["loss_alpha"]
+
+                optimizer_critic.zero_grad()
+                q_loss.backward()
+                optimizer_critic.step()
+                q_losses.append(q_loss.item())
+                optimizer_actor.zero_grad()
+                actor_loss.backward()
+                optimizer_actor.step()
+                actor_losses.append(actor_loss.item())
+                optimizer_alpha.zero_grad()
+                alpha_loss.backward()
+                optimizer_alpha.step()
+                alpha_losses.append(alpha_loss.item())
+
+                # update qnet_target params
+                target_net_updater.step()
+
+                # update priority
+                if cfg.prb:
+                    replay_buffer.update_priority(sampled_tensordict)
+
+        rewards.append(
+            (i, tensordict["next", "reward"].sum().item() / cfg.env_per_collector)
+        )
+        train_log = {
+            "train_reward": rewards[-1][1],
+            "collected_frames": collected_frames,
+        }
+        if q_loss is not None:
+            train_log.update(
+                {
+                    "actor_loss": np.mean(actor_losses),
+                    "q_loss": np.mean(q_losses),
+                    "alpha_loss": np.mean(alpha_losses),
+                    "alpha": loss_td["alpha"],
+                    "entropy": loss_td["entropy"],
+                }
+            )
+        for key, value in train_log.items():
+            logger.log_scalar(key, value, step=collected_frames)
+        if (
+            abs(collected_frames % cfg.eval_iter)
+            < cfg.frames_per_batch * cfg.frame_skip
+        ):
+            with set_exploration_type(ExplorationType.MEAN), torch.no_grad():
```

Comment:
MEAN is usually a bit slower than MODE
MODE just requires one sample but MEAN can require a lot since the average does not in general have a closed form (ie we take the empirical mean).
Not against it but be aware that this will be slower